O365WebHook/o365content.js

```diff
@@ -37,25 +37,40 @@ module.exports.processNotifications = function(context, notifications, callback)
         } else {
             const flattenResult = [].concat.apply([], mapResult);
             context.log.verbose('Messages fetched:', flattenResult.length);
-            return processContent(context, flattenResult, callback);
+            return processContent(context, flattenResult, collector, callback);
         }
     });
 };
 
-function processContent(context, content, callback) {
+function processContent(context, content, collector, callback) {
     const slices = getSliceIndexes(content.length);
-    return async.map(slices, function(slice, asyncCallback){
-        const contentSlice = content.slice(slice.start, slice.end);
-        parseContent(context, contentSlice,
-            function(err, parsedContent) {
-                if (err) {
-                    return asyncCallback(err);
-                }
-                else {
-                    return sendToIngest(context, parsedContent, asyncCallback);
+    return async.reduce(slices, {processed: 0, skip:0},
```

Comment:
Using `reduce` instead of `map` may have a performance impact, i.e. content retrieval will take longer which will cause increased function invocation duration.
I'd suggest to replace `map` with `mapLimit`.